packages/lexical/src/LexicalNormalization.ts

```diff
@@ -22,10 +22,16 @@ function $canSimpleTextNodesBeMerged(
   const node2Mode = node2.__mode;
   const node2Format = node2.__format;
   const node2Style = node2.__style;
+  const node1Classes = node1.__classes || {};
+  const node2Classes = node2.__classes || {};
   return (
     (node1Mode === null || node1Mode === node2Mode) &&
     (node1Format === null || node1Format === node2Format) &&
-    (node1Style === null || node1Style === node2Style)
+    (node1Style === null || node1Style === node2Style) &&
+    Object.keys(node1Classes).length === Object.keys(node2Classes).length &&
+    Object.keys(node1Classes).every(
+      (key) => node1Classes[key] === node2Classes[key],
+    )
```

Comment:
Checking the equivalence between to classes objects should really be in a function where it's easier to early exit and use a more performant for loop instead of `every`

```suggestion
    (node1ClassKeys === null ||
      (node1ClassKeys.length === Object.keys(node2Classes).length &&
        node1ClassKeys.every((key) => node1Classes[key] === node2Classes[key])))
```

Reply:
`Array.every` exits early if it finds a counterexample